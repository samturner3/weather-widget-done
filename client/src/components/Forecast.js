import React from 'react';
// import PropTypes from 'prop-types';
import ReactLoading from 'react-loading';
import moment from 'moment';
import WeatherIcon from 'react-icons-weather';

export default class Forecast extends React.Component {
	Capitalize = (str) => {
		return str.charAt(0).toUpperCase() + str.slice(1);
	}

	render() {
		if (this.props.weatherLoading) {
			return (
				<div className='widget_right_item'>
					<ReactLoading type='spin' height='40px' width='40px' />
				</div>
			);
		} else {
			return (
				this.props.fiveDayForecast.map((forecast, index) => {
					return <div className='widget_right_item' key={index}>
						<div className='widget_right_item_topRow'>
							<div className='widget_right_item-description'>
								{this.Capitalize(forecast.description)} <WeatherIcon name='owm' iconId={forecast.id.toString()} />
							</div>
							<div className='widget_right_item-temp'>
								{Math.round(forecast.temp)}Â°C
							</div>
						</div>
						<div className='widget_right_item_bottomRow'>
							<div className='widget_right_item-day'>
								{moment(forecast.time).format('dddd')}
							</div>
						</div>
					</div>
				})

			);
		}
	}
}
