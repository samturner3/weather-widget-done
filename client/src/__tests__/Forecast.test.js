import React from 'react';
import Forecast from '../components/Forecast';

import { fiveDayForecastMock } from './__mocks__/mockData';

function setup(weatherLoading = false) {
	const props = {
		fiveDayForecast: fiveDayForecastMock,
		weatherLoading
	};

	return {
		wrapper: shallow(<Forecast {...props} />),
		props
	};
}

describe('<Forecast />', () => {
	it('should render the weather info of all five days provided', () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.widget_right_item')).toHaveLength(fiveDayForecastMock.length);
		expect(wrapper.find('WeatherIcon')).toHaveLength(fiveDayForecastMock.length);
		expect(wrapper.find('Loading')).toHaveLength(0);
	});
	it('should render the loader when weather data is loading', () => {
		const { wrapper } = setup(true);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.widget_right_item')).toHaveLength(1);
		expect(wrapper.find('Loading')).toHaveLength(1);
	});
});
