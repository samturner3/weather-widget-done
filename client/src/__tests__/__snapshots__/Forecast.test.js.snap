// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Forecast /> should render the loader when weather data is loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Forecast
    fiveDayForecast={
        Array [
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 10.85,
              "temp_max": 10.85,
              "temp_min": 10.84,
              "time": "2018-08-27 12:00:00",
            },
            Object {
              "description": "broken clouds",
              "id": 803,
              "temp": 10.11,
              "temp_max": 10.11,
              "temp_min": 10.11,
              "time": "2018-08-28 12:00:00",
            },
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 8.55,
              "temp_max": 8.55,
              "temp_min": 8.55,
              "time": "2018-08-29 12:00:00",
            },
            Object {
              "description": "overcast clouds",
              "id": 804,
              "temp": 13.46,
              "temp_max": 13.46,
              "temp_min": 13.46,
              "time": "2018-08-30 12:00:00",
            },
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 15.18,
              "temp_max": 15.18,
              "temp_min": 15.18,
              "time": "2018-08-31 12:00:00",
            },
          ]
    }
    weatherLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loading
        color="#fff"
        delay={0}
        height="40px"
        type="spin"
        width="40px"
/>,
      "className": "widget_right_item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#fff",
        "delay": 0,
        "height": "40px",
        "type": "spin",
        "width": "40px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loading
          color="#fff"
          delay={0}
          height="40px"
          type="spin"
          width="40px"
/>,
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#fff",
          "delay": 0,
          "height": "40px",
          "type": "spin",
          "width": "40px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<Forecast /> should render the weather info of all five days provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Forecast
    fiveDayForecast={
        Array [
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 10.85,
              "temp_max": 10.85,
              "temp_min": 10.84,
              "time": "2018-08-27 12:00:00",
            },
            Object {
              "description": "broken clouds",
              "id": 803,
              "temp": 10.11,
              "temp_max": 10.11,
              "temp_min": 10.11,
              "time": "2018-08-28 12:00:00",
            },
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 8.55,
              "temp_max": 8.55,
              "temp_min": 8.55,
              "time": "2018-08-29 12:00:00",
            },
            Object {
              "description": "overcast clouds",
              "id": 804,
              "temp": 13.46,
              "temp_max": 13.46,
              "temp_min": 13.46,
              "time": "2018-08-30 12:00:00",
            },
            Object {
              "description": "clear sky",
              "id": 800,
              "temp": 15.18,
              "temp_max": 15.18,
              "temp_min": 15.18,
              "time": "2018-08-31 12:00:00",
            },
          ]
    }
    weatherLoading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="widget_right_item_topRow"
>
          <div
                    className="widget_right_item-description"
          >
                    Clear sky
                     
                    <WeatherIcon
                              iconId="800"
                              name="owm"
                    />
          </div>
          <div
                    className="widget_right_item-temp"
          >
                    11
                    °C
          </div>
</div>,
        <div
          className="widget_right_item_bottomRow"
>
          <div
                    className="widget_right_item-day"
          >
                    Monday
          </div>
</div>,
      ],
      "className": "widget_right_item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="widget_right_item-description"
>
              Clear sky
               
              <WeatherIcon
                            iconId="800"
                            name="owm"
              />
</div>,
            <div
              className="widget_right_item-temp"
>
              11
              °C
</div>,
          ],
          "className": "widget_right_item_topRow",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Clear sky",
                " ",
                <WeatherIcon
                  iconId="800"
                  name="owm"
/>,
              ],
              "className": "widget_right_item-description",
            },
            "ref": null,
            "rendered": Array [
              "Clear sky",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "iconId": "800",
                  "name": "owm",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                11,
                "°C",
              ],
              "className": "widget_right_item-temp",
            },
            "ref": null,
            "rendered": Array [
              11,
              "°C",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="widget_right_item-day"
>
            Monday
</div>,
          "className": "widget_right_item_bottomRow",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Monday",
            "className": "widget_right_item-day",
          },
          "ref": null,
          "rendered": "Monday",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="widget_right_item_topRow"
>
            <div
                        className="widget_right_item-description"
            >
                        Clear sky
                         
                        <WeatherIcon
                                    iconId="800"
                                    name="owm"
                        />
            </div>
            <div
                        className="widget_right_item-temp"
            >
                        11
                        °C
            </div>
</div>,
          <div
            className="widget_right_item_bottomRow"
>
            <div
                        className="widget_right_item-day"
            >
                        Monday
            </div>
</div>,
        ],
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget_right_item-description"
>
                Clear sky
                 
                <WeatherIcon
                                iconId="800"
                                name="owm"
                />
</div>,
              <div
                className="widget_right_item-temp"
>
                11
                °C
</div>,
            ],
            "className": "widget_right_item_topRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Clear sky",
                  " ",
                  <WeatherIcon
                    iconId="800"
                    name="owm"
/>,
                ],
                "className": "widget_right_item-description",
              },
              "ref": null,
              "rendered": Array [
                "Clear sky",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconId": "800",
                    "name": "owm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  11,
                  "°C",
                ],
                "className": "widget_right_item-temp",
              },
              "ref": null,
              "rendered": Array [
                11,
                "°C",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget_right_item-day"
>
              Monday
</div>,
            "className": "widget_right_item_bottomRow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Monday",
              "className": "widget_right_item-day",
            },
            "ref": null,
            "rendered": "Monday",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="widget_right_item_topRow"
>
            <div
                        className="widget_right_item-description"
            >
                        Broken clouds
                         
                        <WeatherIcon
                                    iconId="803"
                                    name="owm"
                        />
            </div>
            <div
                        className="widget_right_item-temp"
            >
                        10
                        °C
            </div>
</div>,
          <div
            className="widget_right_item_bottomRow"
>
            <div
                        className="widget_right_item-day"
            >
                        Tuesday
            </div>
</div>,
        ],
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget_right_item-description"
>
                Broken clouds
                 
                <WeatherIcon
                                iconId="803"
                                name="owm"
                />
</div>,
              <div
                className="widget_right_item-temp"
>
                10
                °C
</div>,
            ],
            "className": "widget_right_item_topRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Broken clouds",
                  " ",
                  <WeatherIcon
                    iconId="803"
                    name="owm"
/>,
                ],
                "className": "widget_right_item-description",
              },
              "ref": null,
              "rendered": Array [
                "Broken clouds",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconId": "803",
                    "name": "owm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  10,
                  "°C",
                ],
                "className": "widget_right_item-temp",
              },
              "ref": null,
              "rendered": Array [
                10,
                "°C",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget_right_item-day"
>
              Tuesday
</div>,
            "className": "widget_right_item_bottomRow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tuesday",
              "className": "widget_right_item-day",
            },
            "ref": null,
            "rendered": "Tuesday",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="widget_right_item_topRow"
>
            <div
                        className="widget_right_item-description"
            >
                        Clear sky
                         
                        <WeatherIcon
                                    iconId="800"
                                    name="owm"
                        />
            </div>
            <div
                        className="widget_right_item-temp"
            >
                        9
                        °C
            </div>
</div>,
          <div
            className="widget_right_item_bottomRow"
>
            <div
                        className="widget_right_item-day"
            >
                        Wednesday
            </div>
</div>,
        ],
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget_right_item-description"
>
                Clear sky
                 
                <WeatherIcon
                                iconId="800"
                                name="owm"
                />
</div>,
              <div
                className="widget_right_item-temp"
>
                9
                °C
</div>,
            ],
            "className": "widget_right_item_topRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Clear sky",
                  " ",
                  <WeatherIcon
                    iconId="800"
                    name="owm"
/>,
                ],
                "className": "widget_right_item-description",
              },
              "ref": null,
              "rendered": Array [
                "Clear sky",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconId": "800",
                    "name": "owm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  9,
                  "°C",
                ],
                "className": "widget_right_item-temp",
              },
              "ref": null,
              "rendered": Array [
                9,
                "°C",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget_right_item-day"
>
              Wednesday
</div>,
            "className": "widget_right_item_bottomRow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wednesday",
              "className": "widget_right_item-day",
            },
            "ref": null,
            "rendered": "Wednesday",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "3",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="widget_right_item_topRow"
>
            <div
                        className="widget_right_item-description"
            >
                        Overcast clouds
                         
                        <WeatherIcon
                                    iconId="804"
                                    name="owm"
                        />
            </div>
            <div
                        className="widget_right_item-temp"
            >
                        13
                        °C
            </div>
</div>,
          <div
            className="widget_right_item_bottomRow"
>
            <div
                        className="widget_right_item-day"
            >
                        Thursday
            </div>
</div>,
        ],
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget_right_item-description"
>
                Overcast clouds
                 
                <WeatherIcon
                                iconId="804"
                                name="owm"
                />
</div>,
              <div
                className="widget_right_item-temp"
>
                13
                °C
</div>,
            ],
            "className": "widget_right_item_topRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Overcast clouds",
                  " ",
                  <WeatherIcon
                    iconId="804"
                    name="owm"
/>,
                ],
                "className": "widget_right_item-description",
              },
              "ref": null,
              "rendered": Array [
                "Overcast clouds",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconId": "804",
                    "name": "owm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  13,
                  "°C",
                ],
                "className": "widget_right_item-temp",
              },
              "ref": null,
              "rendered": Array [
                13,
                "°C",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget_right_item-day"
>
              Thursday
</div>,
            "className": "widget_right_item_bottomRow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Thursday",
              "className": "widget_right_item-day",
            },
            "ref": null,
            "rendered": "Thursday",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    Object {
      "instance": null,
      "key": "4",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="widget_right_item_topRow"
>
            <div
                        className="widget_right_item-description"
            >
                        Clear sky
                         
                        <WeatherIcon
                                    iconId="800"
                                    name="owm"
                        />
            </div>
            <div
                        className="widget_right_item-temp"
            >
                        15
                        °C
            </div>
</div>,
          <div
            className="widget_right_item_bottomRow"
>
            <div
                        className="widget_right_item-day"
            >
                        Friday
            </div>
</div>,
        ],
        "className": "widget_right_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="widget_right_item-description"
>
                Clear sky
                 
                <WeatherIcon
                                iconId="800"
                                name="owm"
                />
</div>,
              <div
                className="widget_right_item-temp"
>
                15
                °C
</div>,
            ],
            "className": "widget_right_item_topRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Clear sky",
                  " ",
                  <WeatherIcon
                    iconId="800"
                    name="owm"
/>,
                ],
                "className": "widget_right_item-description",
              },
              "ref": null,
              "rendered": Array [
                "Clear sky",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "iconId": "800",
                    "name": "owm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  15,
                  "°C",
                ],
                "className": "widget_right_item-temp",
              },
              "ref": null,
              "rendered": Array [
                15,
                "°C",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget_right_item-day"
>
              Friday
</div>,
            "className": "widget_right_item_bottomRow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Friday",
              "className": "widget_right_item-day",
            },
            "ref": null,
            "rendered": "Friday",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
